tests/test_concurrency.py

```diff
@@ -0,0 +1,31 @@
+import sys
+
+import pytest
+
+from httpx import AsyncioBackend, HTTPVersionConfig, SSLConfig, TimeoutConfig
+
+
+@pytest.mark.xfail(
+    sys.version_info < (3, 7),
+    reason="Requires Python 3.7+ for AbstractEventLoop.start_tls()",
+)
+@pytest.mark.asyncio
+async def test_start_tls_on_socket_stream(https_server):
+    """
+    See that the backend can make a connection without TLS then
+    start TLS on an existing connection.
+    """
+    backend = AsyncioBackend()
+    ctx = SSLConfig().load_ssl_context_no_verify(HTTPVersionConfig())
+    timeout = TimeoutConfig(5)
+
+    stream = await backend.connect("127.0.0.1", 8001, None, timeout)
+    assert stream.is_connection_dropped() is False
+    assert stream.stream_writer.get_extra_info("cipher", default=None) is None
+
+    stream = await backend.start_tls(stream, "127.0.0.1", ctx, timeout)
+    assert stream.is_connection_dropped() is False
+    assert stream.stream_writer.get_extra_info("cipher", default=None) is not None
```

Comment:
Looks like a good way to check that we're using TLS. Could this be turned into a new method on `ConcurrencyBackend`, though, e.g. `.is_encrypted()`, or `.has_tls()`?

Reply:
I know we've got a lot of effort going on for generalizing test cases but i think we should keep the low level test cases from deciding our ConcurrencyBackend API. What do you think?

Reply:
Okay. I suppose we’ll be able to parametrize how to perform this operation within the test file without touching the concurrency backend anyway. :-)